src/an_sso.coffee

```diff
@@ -322,17 +321,42 @@ window.AsmodeeNet = (->
         oauth(options)
 
     identity: (options) ->
+        if !this.isConnected()
+            console.error  'identity error', 'You\'re not connected'
+            options.error('Identity error. Not connected', null, null, 'Not Connected') if options && options.error
```

Comment:
I'd rather use `console.error()` as a fallback if there is no other facility available, i.e.:

~~~coffeescript
if options && options.error
  options.error(...)
else
  console.error(...)
~~~

Also, you can collapse tests in Coffescript like so:

~~~coffeescript
if foo?.bar?.baz
~~~

This will test each object in the chain individually.